src/check_clickhouse.py

```diff
@@ -0,0 +1,500 @@
+    )
+    parser_replication.add_argument(
+        '--merges-in-queue', type=int, default=40,
+        help='how many merges are delayed, could be greater then 0 for a long',
+    )
+    parser_replication.add_argument(
+        '--log-delay', type=int, default=10, help='how big difference between '
+        'actions log in ZK and locally could be',
+    )
+    parser_replication.add_argument(
+        '--absolute-delay', type=int, default=40,
+        help='how big lag in seconds the current replica could have',
+    )
+
+
+class Check(object):
+    def __init__(self, *args, **kwargs):
+        self.conn = Client(*args, **kwargs)
+        self.code = Code()
+
+    def execute(self, *args, **kwargs) -> List[tuple]:
+        """
+        Executes ClickHouse SQL with catching of NetworkError as CRITICAL
+        """
+        return self.conn.execute(*args, **kwargs)
+
+    def execute_dict(self, *args, **kwargs) -> List[dict]:
+        """
+        Wrapper around execute() to return list of rows as dict
+        """
+        kwargs['with_column_types'] = True
+        rows, columns = self.execute(*args, **kwargs)
+        result = [{columns[i][0]: v for i, v in enumerate(r)} for r in rows]
+        return result
+
+    def exit(self, message: str) -> ExitStruct:
+        message = self.code.name + ': ' + message
+        return self.code.current, message
+
+
+class CheckClusters(Check):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __call__(self, config: dict) -> ExitStruct:
+        """
+        Accepts args with 'clusters' attribute and checks every related
+        Distributed table, if it is readable
+        """
+        logger.debug('Cluster check, config={}'.format(pformat(config)))
+        messages = []  # type: List[str]
+        # Request config['clusters'] from ClickHouse
+        clusters = self.execute_dict(
+            r'''
+            SELECT
+                splitByChar('\'', engine_full)[2] AS cluster,
+                groupArray(concat(t.database, '.', t.name)) AS tables
+            FROM system.tables AS t
+            INNER JOIN
+            (
+                SELECT cluster
+                FROM system.clusters
+                WHERE cluster IN %(clusters)s
+                GROUP BY cluster
+            ) AS c USING (cluster)
+            WHERE t.engine = 'Distributed'
+            GROUP BY cluster
+            ''',
+            {'clusters': tuple(config['clusters'])},
+        )
+        logger.debug(
+            'Existing clusters on ClickHouse server are: {}'
+            .format(pformat(clusters))
+        )
+
+        existing = {c['cluster'] for c in clusters}
+        missing = set(config['clusters']) - existing
+        if missing:
+            self.code.current = Code.CRITICAL
+            messages.append('Clusters not found on server: {}'.format(missing))
+
+        def checkTables(tables: List[str]) -> List[str]:
```

Comment:
Sry, actually my bad as it was in the suggestion. Too many switches between programming languages. `pascalCase` for methods is not pythonic, so:
```suggestion
        def check_tables(tables: List[str]) -> List[str]:
```

Reply:
Done in another commit.